centreon/www/front_src/src/CloudNotificationsConfiguration/api/useDeleteRequest.tsx

```diff
@@ -17,83 +18,106 @@ import {
+    : labelNotificationsSuccessfullyDeleted;
+  const payload = isSingleItem ? {} : { ids: deleteNotification.id };
+
   const { isMutating, mutateAsync } = useMutationQuery({
     defaultFailureMessage: t(labelFailed) as string,
-    getEndpoint,
+    getEndpoint: (): string => endpoint,
     method: fetchMethod
   });
 
-  const onConfirm = (): void => {
-    mutateAsync(payload || {}).then((response) => {
-      const { isError, statusCode, message, data } = response as ResponseError;
+  const submit = (): void => {
+    mutateAsync(payload || {})
+      .then((response) => {
+        const { isError, statusCode, message, data } =
+          response as ResponseError;
 
-      if (isError) {
-        return;
-      }
+        if (isError) {
+          return;
+        }
 
-      if (statusCode === 207) {
-        const successfullResponses = data.filter(propEq('status', 204));
-        const failedResponsesIds = data
-          .filter(complement(propEq('status', 204)))
-          .map((item) => item.href)
-          .map((item) => parseInt(last(split('/', item)) as string, 10));
+        if (statusCode === 207) {
+          const successfullResponses = data.filter(propEq('status', 204));
+          const failedResponsesIds = data
+            .filter(complement(propEq('status', 204)))
+            .map((item) => item.href)
+            .map((item) => parseInt(last(split('/', item)) as string, 10));
 
-        if (isEmpty(successfullResponses)) {
-          showErrorMessage(t(labelFailed));
-          setDialogOpen?.(false);
+          if (isEmpty(successfullResponses)) {
+            showErrorMessage(t(labelFailed));
 
-          return;
-        }
+            return;
+          }
 
-        if (length(successfullResponses) < length(data)) {
-          const failedResponsesName = selectedRows
-            ?.filter((item) => includes(item.id, failedResponsesIds))
-            .map((item) => item.name);
+          if (length(successfullResponses) < length(data)) {
+            const failedResponsesName = selectedRows
+              ?.filter((item) => includes(item.id, failedResponsesIds))
+              .map((item) => item.name);
 
-          showWarningMessage(
-            `${labelFailedToDeleteNotifications}: ${failedResponsesName.join(
-              ', '
-            )}`
-          );
-          setDialogOpen?.(false);
+            showWarningMessage(
+              `${labelFailedToDeleteNotifications}: ${failedResponsesName.join(
+                ', '
+              )}`
+            );
 
-          return;
+            return;
+          }
         }
-      }
 
-      showSuccessMessage(message || t(labelSuccess));
-      setDialogOpen?.(false);
-      onSuccess?.();
-      queryClient.invalidateQueries(['notifications']);
-    });
+        showSuccessMessage(message || t(labelSuccess));
+        queryClient.invalidateQueries(['notifications']);
+      })
+      .finally(() => {
+        closeDialog();
+        closePanel();
+      });
```

Comment:
separations of concerns, imo. this should be the responsibility of the `useDelete` hook, not the data request hook...